File: src/main/scala/robosky/ether/block/machine/infuser/InfuserContainer.scala
package robosky.ether.block.machine.infuser

import io.github.cottonmc.cotton.gui.CottonScreenController
import io.github.cottonmc.cotton.gui.widget.{WGridPanel, WItemSlot, WLabel}
import net.minecraft.container.BlockContext
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.text.TranslatableText
import robosky.ether.block.machine.RecipeRegistry

class InfuserContainer(syncId: Int, playerInventory: PlayerInventory, ctx: BlockContext)
  extends CottonScreenController(RecipeRegistry.aegisaltRecipe, syncId, playerInventory,
    CottonScreenController.getBlockInventory(ctx), CottonScreenController.getBlockPropertyDelegate(ctx)) {

  private val be: AegisaltInfuser =
    ctx.run((w, p) => w.getBlockEntity(p).asInstanceOf[AegisaltInfuser]).get()

  def init() {
    val gridRoot = rootPanel.asInstanceOf[WGridPanel]
    gridRoot.add(
      new WLabel(
        new TranslatableText("block.luminiferous_uplands.aegisalt_infuser"),
        WLabel.DEFAULT_TEXT_COLOR
      ),
      0,
      0
    )

    gridRoot.add(WItemSlot.of(blockInventory, 0, 2, 1), 2, 1)
    gridRoot.add(WItemSlot.of(blockInventory, 2), 5, 2)
    gridRoot.add(WItemSlot.outputOf(blockInventory, 3), 7, 1)
    gridRoot.add(this.createPlayerInventoryPanel(), 0, 4)

    gridRoot.validate(this)
  }

  init()

  override def getCraftingResultSlotIndex: Int = -1
}

File: src/main/java/robosky/uplands/block/machine/infuser/InfuserContainer.java
package robosky.uplands.block.machine.infuser;

import io.github.cottonmc.cotton.gui.SyncedGuiDescription;
import io.github.cottonmc.cotton.gui.widget.WGridPanel;
import io.github.cottonmc.cotton.gui.widget.WItemSlot;
import io.github.cottonmc.cotton.gui.widget.WLabel;
import robosky.uplands.block.machine.MachineRegistry;

import net.minecraft.entity.player.PlayerInventory;
import net.minecraft.screen.ScreenHandlerContext;
import net.minecraft.text.TranslatableText;

public class InfuserContainer extends SyncedGuiDescription {

    public InfuserContainer(int syncId, PlayerInventory playerInventory, ScreenHandlerContext ctx) {
        super(MachineRegistry.AEGISALT_INFUSER.gui, syncId, playerInventory,
            SyncedGuiDescription.getBlockInventory(ctx), SyncedGuiDescription.getBlockPropertyDelegate(ctx));
        init();
    }

    private void init() {
        WGridPanel gridRoot = (WGridPanel)rootPanel;
        gridRoot.add(
            new WLabel(
                new TranslatableText("block.luminiferous_uplands.aegisalt_infuser"),
                WLabel.DEFAULT_TEXT_COLOR
            ),
            0,
            0
        );

        gridRoot.add(WItemSlot.of(blockInventory, 0, 2, 1), 2, 1);
        gridRoot.add(WItemSlot.of(blockInventory, 2), 5, 2);
        gridRoot.add(WItemSlot.outputOf(blockInventory, 3), 7, 1);
        gridRoot.add(this.createPlayerInventoryPanel(), 0, 4);

        gridRoot.validate(this);
    }

    @Deprecated
    public int getCraftingResultSlotIndex() {
        return -1;
    }
}
